packages/web-app-files/tests/unit/components/SideBar/Links/PrivateLinkItem.spec.js

```diff
@@ -0,0 +1,116 @@
+import Vuex from 'vuex'
+import GetTextPlugin from 'vue-gettext'
+import DesignSystem from 'owncloud-design-system'
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import PrivateLinkItem from '@files/src/components/SideBar/Links/PrivateLinkItem.vue'
+
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const selectors = {
+  linkItem: '.oc-files-private-link-item',
+  copyToClipboardButton: '.oc-files-private-link-copy-url'
+}
+
+global.window = Object.create(window)
+const windowLocation = { href: 'http://localhost:9100' }
+Object.defineProperty(window, 'location', {
+  value: windowLocation
+})
+
+describe('PrivateLinkItem', () => {
+  function createStore({ isMounted = true } = {}) {
+    return new Vuex.Store({
+      getters: {
+        capabilities: function() {
+          return {
+            files: {
+              privateLinks: true
+            }
+          }
+        }
+      },
+      modules: {
+        Files: {
+          namespaced: true,
+          getters: {
+            highlightedFile: function() {
+              return {
+                isMounted: jest.fn(() => isMounted),
+                name: 'lorem.txt',
+                privateLink: 'some-link'
+              }
+            }
+          }
+        }
+      }
+    })
+  }
+
+  function getShallowWrapper(store) {
+    return shallowMount(PrivateLinkItem, {
+      localVue,
+      store,
+      stubs: {
+        'copy-to-clipboard-button': true
+      }
+    })
+  }
+
+  describe('when private link is enabled', () => {
```

Comment:
```suggestion
  describe('when the private link function is enabled', () => {
```

Reply:
```suggestion
  describe('when the private link functionality is enabled', () => {
```

I am not 100%  sure what to say here. In the context of computer code, a "function" is a special thing that takes input parameters and has a return value. IMO this text is not talking about that sort of "function", but about "private link functionality"?

Reply:
why function? it is an attribute of `file.capabilities`

Reply:
functionality not function, that attribute enables a specific functionality

Reply:
fixed